pkg/operator/controllers/base/aro_controller_test.go

```diff
@@ -43,29 +44,92 @@ func TestSetConditions(t *testing.T) {
+		name   string
+		start  []operatorv1.OperatorCondition
+		action func(c AROController)
+		want   []operatorv1.OperatorCondition
 	}{
 		{
-			name:  "sets all provided conditions",
+			name:  "SetConditions - sets all provided conditions",
 			start: []operatorv1.OperatorCondition{internetReachable},
-			input: []*operatorv1.OperatorCondition{&defaultAvailable, &defaultProgressing, &defaultDegraded},
-			want:  []operatorv1.OperatorCondition{internetReachable, defaultAvailable, defaultProgressing, defaultDegraded},
+			action: func(c AROController) {
+				c.SetConditions(ctx, &defaultAvailable, &defaultProgressing, &defaultDegraded)
+			},
+			want: []operatorv1.OperatorCondition{internetReachable, defaultAvailable, defaultProgressing, defaultDegraded},
 		},
 		{
-			name:  "if condition exists and status matches, does not update",
+			name:  "SetConditions - if condition exists and status matches, does not update",
 			start: []operatorv1.OperatorCondition{internetReachable, defaultAvailableInPast},
-			input: []*operatorv1.OperatorCondition{&defaultAvailable, &defaultProgressing, &defaultDegraded},
-			want:  []operatorv1.OperatorCondition{internetReachable, defaultAvailableInPast, defaultProgressing, defaultDegraded},
+			action: func(c AROController) {
+				c.SetConditions(ctx, &defaultAvailable, &defaultProgressing, &defaultDegraded)
+			},
+			want: []operatorv1.OperatorCondition{internetReachable, defaultAvailableInPast, defaultProgressing, defaultDegraded},
 		},
 		{
-			name:  "if condition exists and status does not match, updates",
+			name:  "SetConditions - if condition exists and status does not match, updates",
 			start: []operatorv1.OperatorCondition{internetReachable, defaultAvailableInPast},
-			input: []*operatorv1.OperatorCondition{&unavailable, &defaultProgressing, &defaultDegraded},
-			want:  []operatorv1.OperatorCondition{internetReachable, unavailable, defaultProgressing, defaultDegraded},
+			action: func(c AROController) {
+				c.SetConditions(ctx, &unavailable, &defaultProgressing, &defaultDegraded)
+			},
+			want: []operatorv1.OperatorCondition{internetReachable, unavailable, defaultProgressing, defaultDegraded},
+		},
+		{
+			name:  "SetProgressing - sets Progressing to true with message",
+			start: []operatorv1.OperatorCondition{defaultAvailable, defaultProgressing, defaultDegraded},
+			action: func(c AROController) {
+				c.SetProgressing(ctx, isProgressing.Message)
+			},
+			want: []operatorv1.OperatorCondition{defaultAvailable, isProgressing, defaultDegraded},
+		},
+		{
+			name:  "ClearProgressing - sets Progressing to false and clears message",
+			start: []operatorv1.OperatorCondition{defaultAvailable, isProgressing, defaultDegraded},
+			action: func(c AROController) {
+				c.ClearProgressing(ctx)
+			},
+			want: []operatorv1.OperatorCondition{defaultAvailable, defaultProgressing, defaultDegraded},
+		},
+		{
+			name:  "SetProgressing - sets Progressing to true with message",
```

Comment:
It looks like this is a duplicate of the `SetProgressing` test case two list entries up. Same thing for the `ClearProgressing` test case directly below this one.

Reply:
good catch, thanks